### DISCUSSION TITLE
Generating double extended keys

### DISCUSSION THREAD

**Discussion by ghost** [trustwallet/wallet-core#2135] [Q&A] [answered]
guard let wallet = HDWallet(mnemonic: mnemonic, passphrase: pin) else {
      throw "New base wallet could not be generated"
    }
    
    let walletKey: PrivateKey = wallet.getKeyForCoin(coin: .cardano)

The code above is how I am currently generating private keys for Cardano, however, these keys produce an ``"Invalid extended key"`` error when signing transactions. Any idea of what I'm doing wrong?

**Comment by optout21** [comment, marked as answer]
I think this is a bug.  For cardano we use 'double extended' private key, which has 6 32-byte parts.  However, through the C (and thus Swift/Kotlin) interfaces (TWPrivateKey), only the first 32-bytes can be accessed.
Created issue #2136 .

**Reply by optout21** [reply]
Please retest, fix has been merged  https://github.com/trustwallet/wallet-core/pull/2142

**Reply by ghost** [reply]
Updates from #2142 still produce an error when signing txs with 12, 15, and 24 word mnemonics. These suspicions were confirmed after running the swift tests locally:

Produced key = "e8c8c5b2df13f3abed4e6b1609c808e08ff959d7e6fc3d849e3f2880550b5744"

Expected key = "e8c8c5b2df13f3abed4e6b1609c808e08ff959d7e6fc3d849e3f2880550b574437aa559095324d78459b9bb2da069da32337e1cc5da78f48e1bd084670107f3110f3245ddf9132ecef98c670272ef39c03a232107733d4a1d28cb53318df26fae0d152bb611cb9ff34e945e4ff627e6fba81da687a601a879759cd76530b5744424db69a75edd4780a5fbc05d1a3c84ac4166ff8e424808481dd8e77627ce5f5bf2eea84515a4e16c4ff06c92381822d910b5cbf9e9c144e1fb76a6291af7276"

I downloaded the package from master, cleaned package and build caches, and can see the updates within my local package. If you know why I am still experiencing this error, please let me know!

**Reply by optout21** [reply]
Most likely running pre-fix version, maybe because it you are dowloading latest package, and not building it locally: there has not been a package release yet since yesterday (@hewigovens?)

**Reply by hewigovens** [reply]
I haven't made a new release yet, will make one pre release today

**Reply by hewigovens** [reply]
Please try downloading 2.7.8 from release page, we don't always update master

**Reply by ghost** [reply]
Just downloaded 2.7.8 and it still doesn't work. The Package.swift is still pointing to 2.7.7

**Reply by ghost** [reply]
> Please try downloading 2.7.8 from release page, we don't always update master

Any updates on my last comment?

**Reply by hewigovens** [reply]
Master updated to 2.7.8
